illumos-utils/src/running_zone.rs

```diff
@@ -1020,19 +1020,42 @@ impl RunningZone {
         // See https://illumos.org/man/8/logadm for details on the naming
         // conventions around these files.
         let dir = current_log_file.parent().unwrap();
-        let mut rotated_files = Vec::new();
+        let mut rotated_files: Vec<Utf8PathBuf> = Vec::new();
         for entry in dir.read_dir_utf8()? {
             let entry = entry?;
             let path = entry.path();
-            if path != current_log_file && path.starts_with(&current_log_file) {
-                rotated_files
-                    .push(root.join(path.strip_prefix("/").unwrap_or(path)));
+
+            // Camino's Utf8Path only considers whole path components to match,
+            // so convert both paths into a String and use that object's
+            // starts_with. See the `camino_starts_with_behaviour` test.
+            if path != current_log_file
+                && path.to_string().starts_with(&current_log_file.to_string())
```

Comment:
Since `Utf8PathBuf` implements `AsRef<str>`, I think we can accomplish this without temporaries? Something like:

```rust
let full: &str = path.as_ref();
if path != current_log_file && full.starts_with(current_log_file.as_ref()) {
    ...
}
```

Reply:
Nice! Done in 73989a9bb